sound/soc/sof/sof-audio.h

```diff
@@ -390,6 +390,13 @@ struct snd_sof_widget {
 	int num_source_pins;
 	int num_sink_pins;
 
+	/* The pin binding array, it takes the form of 
+	 * [widget_name_connected_to_pin0, widget_name_connected_to_pin1, ...],
+	 * with the index as the queue ID.
+	 */
+	char *src_pin_binding;
+	char *sink_pin_binding;
```

Comment:
is a char* really good enough here? Don't you need a char **?

Reply:
the idea is packing all the binding string into one string array, so you find them by str + i * binding_string_size

Reply:
We can use `char**` here to denote that src_pin_binding/sink_pin_binding are string arrays, but it will be used just like a `char*`: `str + i * binding_string_size`

Reply:
I prefer `char **` since they are num_source_pins strings instead of a single string.

Reply:
but it make both memory allocation and free complex, and even with `char *`, it is not hard to understand and to use. in addition, I already add comments.

Reply:
> but it make both memory allocation and free complex, and even with `char *`, it is not hard to understand and to use. in addition, I already add comments.

But *src_pin_binding[index] is more readable than `pin_binding + i * SNDRV_CTL_ELEM_ID_NAME_MAXLEN`, right? And you don't need `bind_widget_name `.
If you really want to use a single char array, please add a macro and always use the macro to separate different strings.
Or you can probably use `char *src_pin_binding[SNDRV_CTL_ELEM_ID_NAME_MAXLEN]` as it is a fixed length string and memory allocation and free will not be more complex.
You should add comments wherever src_pin_binding is used not only here or people will get confused.
Same as sink_pin_binding.
Let's wait for comments from other reviewers. @plbossart @ujfalusi @kv2019i